solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+resource networkManager 'Microsoft.Network/networkManagers@2022-05-01' = {
+  name: 'avnm-${location}'
+  location: location
+  properties: {
+    networkManagerScopeAccesses: [
+      'Connectivity'
+      'SecurityAdmin'
+    ]
+    networkManagerScopes: {
+      subscriptions: [
+        '/subscriptions/${subscription().subscriptionId}'
+      ]
+      managementGroups: []
+    }
+  }
+  resource networkGroupProd 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-spokes-prod'
+    properties: {
+      description: 'Prod Spoke VNETs Network Group'
+    }
+    resource staticMembersSpokeOne 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: prodNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembersSpokeTwo 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: prodNetworkGroupMembers[1]
+      }
+    }
+  }
+  resource networkGroupNonProd 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-spokes-nonprod'
+    properties: {
+      description: 'Non-prod Spoke VNETs Network Group'
+    }
+    resource staticMembersSpokeOne 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembersSpokeTwo 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+  }
+  resource networkGroupAll 'networkGroups@2022-05-01' = {
+    name: 'ng-${location}-all'
+    properties: {
+      description: 'All VNETs Network Group (for Security Configurations)'
+    }
+    resource staticMembers1 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-0'
+      properties: {
+        resourceId: prodNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers2 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-1'
+      properties: {
+        resourceId: prodNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers3 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-2'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[0]
+      }
+    }
+    resource staticMembers4 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-3'
+      properties: {
+        resourceId: nonProdNetworkGroupMembers[1]
+      }
+    }
+    resource staticMembers5 'staticMembers@2022-05-01' = {
+      name: 'sm-${location}-4'
+      properties: {
+        resourceId: hubVnetId
+      }
+    }
+  }
+}
+
+@description('This connectivity configuration defines the connectivity between the spokes. Only deployed if requested.')
+resource connectivityConfigurationNonProd 'Microsoft.Network/networkManagers/connectivityConfigurations@2022-05-01' = {
+  name: 'cc-${location}-spokesnonprod'
+  parent: networkManager
```

Comment:
Curious if there is any reason this _should_ be defined as a `resource` with a `parent` instead of being nested above like the `staticMembers` were in parent `Microsoft.Network/networkManagers`  You can still do depends on, etc.  Some other technical reason maybe?

For what it's worth, they do actually show up as nested children resources even in the generated ARM json.

Reply:
This could be nested, but it seemed clearer to declare it separately rather than have interdependent sub-resources of the network manager all together. It's complex either way...

Reply:
Agreed--I'll pull the network group declarations out of the network manager for consistency.